### PR TITLE
Remove cvx and quadprog from controller

### PR DESCRIPTION
**Pull request by iman512003**
This is to address #51.

### CODE REVIEW

File: smarts/core/controllers/trajectory_tracking_controller.py [hunk 1]
```diff
@@ -531,25 +534,20 @@ def MPC(
...
-            matrix(np.eye(matrix_H.shape[0])),
-            matrix(np.ones((matrix_H.shape[0], 1))),
-        )
+        # This is the solution to the unconstrained optimization problem
+        # for the MPC.
+        unconstrained_optimal_solution = np.matmul(
+            np.linalg.inv(2 * matrix_H),
+            np.matmul(matrix_F, np.array([[lateral_error], [0], [heading_error], [0]]))
+            + np.reshape(matrix_F1, (-1, 1)),
+        )[0][0]
```
Comment (davidrusu) [inline smarts/core/controllers/trajectory_tracking_controller.py:551]: that's it? cool

**Review by davidrusu — APPROVED** [review]

File: smarts/core/controllers/trajectory_tracking_controller.py [hunk 2]
```diff
@@ -25,8 +25,6 @@
 import numpy as np
 
 from scipy import signal
-from cvxopt import matrix, solvers
-from qpsolvers import solve_qp
```
Comment (Gamenot) [inline smarts/core/controllers/trajectory_tracking_controller.py:29]: Remember to also remove this from `setup.py` and `requirements.txt`.

**Review by Gamenot — CHANGES_REQUESTED** [review]

File: smarts/core/controllers/trajectory_tracking_controller.py (hunk 2 above)
Comment (liamchzh) [reply smarts/core/controllers/trajectory_tracking_controller.py:29]: There are other usages in `lane_following_controller.py` and `actuator_dynamic_controller.py`. @iman512003 Are you going to eliminate them in one PR?

Comment (iman512003) [reply smarts/core/controllers/trajectory_tracking_controller.py:29]: > Remember to also remove this from `setup.py` and `requirements.txt`.

Thanks, sure.

Comment (iman512003) [reply smarts/core/controllers/trajectory_tracking_controller.py:29]: > There are other usages in `lane_following_controller.py` and `actuator_dynamic_controller.py`. @iman512003 Are you going to eliminate them in one PR?

Oh, Thanks, sure. I removed them from other controller files as well.

File: smarts/core/controllers/trajectory_tracking_controller.py (hunk 1 above)
Comment (iman512003) [reply smarts/core/controllers/trajectory_tracking_controller.py:551]: Thanks, Yes, basically, the MPC formulation results in the following QP:

min (1/2)x^T*H*x+F^T*X, subject to A*X<=B
  x

Solving the above full QP requires handling constraint(something like interior point method) but for unconstrained formulation(i.e ignoring A*X<=B) we can simply take the derivative which results in:

H*x+F=0 hence X=-H^-1 * F
Also, Autoware has a internal unconstrained solver similar to the above solution:

https://github.com/Autoware-AI/core_planning/blob/master/mpc_follower/src/qp_solver/qp_solver_unconstr.cpp

**Review by liamchzh — APPROVED** [review]

**Review by Gamenot — APPROVED** [review]

**Pull request merged**